packages/vitest/src/runtime/loader.ts

```diff
@@ -0,0 +1,64 @@
+import { pathToFileURL } from 'url'
+import { normalizeModuleId } from 'vite-node/utils'
+import type { Awaitable } from '../types'
+import { getWorkerState } from '../utils'
+
+interface ModuleContext {
+  conditions: string[]
+  parentURL?: string
+}
+
+enum ModuleFormat {
+  Builtin = 'builtin',
+  Commonjs = 'commonjs',
+  Json = 'json',
+  Module = 'module',
+  Wasm = 'wasm',
+}
+
+interface ResolveResult {
+  url: string
+  format?: ModuleFormat
+}
+
+interface Resolver {
+  (url: string, context: ModuleContext, next: Resolver): Awaitable<ResolveResult>
+}
+
+interface LoaderContext {
+  format: ModuleFormat
+  importAssertions: Record<string, string>
+}
+
+interface LoaderResult {
+  format: ModuleFormat
+  source: string | ArrayBuffer | SharedArrayBuffer | Uint8Array
+}
+
+interface Loader {
+  (url: string, context: LoaderContext, next: Loader): Awaitable<LoaderResult>
+}
+
+export const resolve: Resolver = async (url, context, next) => {
+  const { parentURL } = context
+  if (!parentURL || !parentURL.includes('node_modules'))
+    return next(url, context, next)
+
+  const id = normalizeModuleId(url)
+  const importer = normalizeModuleId(parentURL)
+  const state = getWorkerState()
```

Comment:
This won't be possible in future versions of loader, because it will be running in a separate thread. Is there a way to communicate between the main thread? @antfu I didn't find a way to pass down `workerPort` to the loader.

Reply:
We can pass down `argv` to the loader, so it would know what to listen for example, but we need to setup something to listen to.

Reply:
I guess we need to count on Node to provide a way for communication? (and I think they should if they move it to a thread)